base/process.py

```diff
@@ -302,6 +315,58 @@ def filter_alignment(aln, region=None, lower_tp=None, upper_tp=None):
                          self.mutation_frequency_counts), fh, protocol=pickle.HIGHEST_PROTOCOL)
 
 
+    def global_frequencies(self, min_freq):
+        # determine sites whose frequencies need to be computed in all regions
+        self.seqs.diversity_statistics()
+        include_set = {}
+        for prot in ['nuc'] + self.seqs.translations.keys():
+            include_set[prot] = np.where(np.sum(self.seqs.af[prot][:-2]**2, axis=0)<np.sum(self.seqs.af[prot][:-2], axis=0)**2-min_freq)[0]
+
+        # set pivots and define groups of larger regions for frequency display
+        pivots = self.get_pivots_via_spacing()
+        acronyms = set([x[1] for x in self.info["regions"] if x[1]!=""])
+        region_groups = {str(x):[str(y[0]) for y in self.info["regions"] if y[1] == x] for x in acronyms}
+        pop_sizes = {str(x):np.sum([y[-1] for y in self.info["regions"] if y[1] == x]) for x in acronyms}
+        total_popsize = np.sum(pop_sizes.values())
+
+        # estimate frequencies in individual regions
+        # TODO: move inertia and stiffness parameters to config
+        for region in region_groups.iteritems():
+            self.estimate_mutation_frequencies(pivots=pivots, region=region, min_freq=0.02, include_set=include_set,
+                                                 inertia=np.exp(-2.0/12), stiffness=2.0*12)
+
+        # perform a weighted average of frequencies across the regions to determine
+        # global frequencies.
+        # First: compute the weights accounting for seasonal variation and populations size
```

Comment:
I get how population size is affecting sampling rate in line 346. I don't really get how seasonality is working in line 345. Are you trying to flatten the seasonal signature? Can't tell what's going on here.

Reply:
I am trying to weigh regions by their seasonal contribution approximated by the sequence counts per pivot. This is normalized to its max via `weights[region]/weights[region].max()`. To prevent an on/off pattern for sparsely sampled regions, I added this floor of 0.1 (rel to max). No good reason for this particular value or functional form, but this is what is first put and it stuck...

Reply:
I was kind of assuming you could just assign each sequence a weight based on population size and then let the seasonal distribution do its thing. For example:

We have two regions `AS` and `NA`. `AS` has 1000 sequences and 2 billion people. `NA` has 2000 sequences and 500 million people. This makes it so that each `AS` sequence has 8x the weight as each `NA` sequence when determining global frequencies.

Assuming `AS` sequences are flat throughout the year and `NA` sequences preferentially appear in the winter. In this case, we'd have, say, 50 `AS` sequences in Jul and 25 `NA` sequences in Jul. Here the Jul pivot would have contributions of 8x 50 from `AS` and 1x 25 from `NA`. And then, we'd have, say, 50 `AS` sequences in Jan and 400 `NA` sequences in Jan. Here the Jan pivot would have contributions of 8x 50 from `AS` and 1x 400 from `NA`.

Reply:
That would be one way of doing it, but currently we are estimating frequencies within regions first (since we want them anyway) and then take a weighted average. Since frequencies are between 0 and 1, differences in sampling effort are already ironed out at that stage. Multiplying by the variation throughout the year accounts for the seasonal variation in the same way as the above (except that one should probably normalize to the mean instead of the max).

Reply:
I see now. Okay. Thank you.